tasking/pulp/tasking/storage.py

```diff
@@ -0,0 +1,123 @@
+import errno
+import os
+import shutil
+import stat
+
+from celery import task
+
+from django.conf import settings as pulp_settings
+
+
+def _working_dir_root(worker_name):
+    """
+    Returns the path to the working directory of a worker
+    :param worker_name:     Name of worker for which path is requested
+    :type  name:            basestring
+    :return working_directory setting from server config
+    :rtype  str
+    """
+    working_dir = pulp_settings.SERVER['working_directory']
+    dir_root = os.path.join(working_dir, worker_name)
+    return dir_root
+
+
+def create_worker_working_directory(worker_name):
+    """
+    Creates a working directory inside the working_directory as specified in /etc/pulp/server.yaml
+    default path for working_directory is /var/cache/pulp
+    :param worker_name:     Name of worker that uses the working directory created
+    :type  name:            basestring
+    :return Path to the working directory for a worker
+    :rtype  str
+    """
+    working_dir_root = _working_dir_root(worker_name)
+    os.mkdir(working_dir_root)
+
+
+def delete_worker_working_directory(worker_name):
+    """
+    Deletes a working directory inside the working_directory as specified in /etc/pulp/server.yaml
```

Comment:
We should probably replace all usage of `/etc/pulp/server.yaml` with `the settings file`. The location could change in the future.